### DISCUSSION TITLE
T-Display S3 Support?

### DISCUSSION THREAD

**Discussion by MaxSMoke777** [moononournation/Arduino_GFX#227] [General]
Any thought about adding T-Display S3 support? If you've never seen one of these, here's a sample: 
https://www.aliexpress.us/item/3256804516036483.html

It seems to be very different from other ESP32 chips. It runs parallel graphics on a gorgeous 170x320 display, so I'd love to use your code with it.

**Comment by moononournation** [comment]
It is already supported: https://youtu.be/kpRC64QNQAo

Please find more details in PDqgraphicstest example

**Reply by MaxSMoke777** [reply]
I was able to get the PDqgraphicstest to work, but I can't get your AnimatedGIF player to function at all. The screen always remains black. What's more, the Arduino serial monitor doesn't work and the board has difficulty flashing while your code is running.

**Reply by moononournation** [reply]
try smaller gif file

**Reply by MaxSMoke777** [reply]
I used your sample image. There isn't even text on the screen that says there's no image if I don't first load one into the SPIFF. It's like the animated GIF example has no setup for the S3. I used this part from the PDqgraphicstest: 

#define TFT_PWD 15
#define TFT_BL 38
Arduino_DataBus *bus = new Arduino_ESP32LCD8(
    7 /* DC */, 6 /* CS */, 8 /* WR */, 9 /* RD */,
    39 /* D0 */, 40 /* D1 */, 41 /* D2 */, 42 /* D3 */, 45 /* D4 */, 46 /* D5 */, 47 /* D6 */, 48 /* D7 */);
Arduino_GFX *gfx = new Arduino_ST7789(bus, 5 /* RST */, 0 /* rotation */, true /* IPS */, 170 /* width */, 320 /* height */, 35 /* col offset 1 */, 0 /* row offset 1 */, 35 /* col offset 2 */, 0 /* row offset 2 */);

But the screen never shows anything. Are you certain you updated your Animated GIF example? I think GitHub has an old example.

**Reply by moononournation** [reply]
please copy and replace the TFT_PWD and TFT_BL related code from PDQGraphicstest to your sketch

**Reply by MaxSMoke777** [reply]
That is what I did. I just told you I did that. You can see what I copied above. 

Have you tested the Animated GIF player yourself with the S3? If you test it yourself, I think you will find it doesn't work. If it does work, I'd like to see the code you used. We are using the same board, so if you can make it work, then I should be able to as well.

**Reply by moononournation** [reply]
You said PDQgraphicstest ok, right? And the GIF example not yet have handle TFT_PWD code, so I ask you copy related code in PDQ main() to GIF example.

**Reply by MaxSMoke777** [reply]
Yes, I used that code. I could not get it to work. Have you tried yourself?

**Reply by moononournation** [reply]
https://user-images.githubusercontent.com/13228891/203873334-a54e275c-bb3a-4c42-b12a-b01f97bf1cd6.MOV

**Reply by MaxSMoke777** [reply]
Could you please show the code you used? I don't see what I'm missing. Or just update the examples in your GitHub? That would help everyone. Thanks!

**Reply by moononournation** [reply]
I know many people mixed up TFT_BL and GFX_BL, I will improve it soon.

**Comment by moononournation** [comment]
Please get update to check is it better now.

**Reply by MaxSMoke777** [reply]
Success! Amazing as always! I tell everybody I can about your library. Thank you!

**Reply by MaxSMoke777** [reply]
I made a few YouTube shorts to show off the library, and linked them all back to your page. Here's my favorite... 
https://www.youtube.com/shorts/-Nn8G77lPZw

**Comment by h5n1xp** [comment]
I've been using Arduino_GFX with a T-Display S3 and it works really well.

I'm using the canvas object as a frambuffer which I "flush()" to the display once per frame redraw. I achieve approximately 26FPS, which is pretty good but it would be nice to have it a bit faster, and seems pushing the data to the ST7789 is the bottleneck (increasing the scene complexity doesn't slow the refresh significantly).

My question is how to determine the speed of the 8bit Parallel bus, and can it be increased?

-edit- I have played with the begin() function and managed to get the display functional up to a speed of 30800000, after which there is graphical corruption or just a blank display.

**Comment by moononournation** [comment]
Try `gfx->begin(40000000);`, it should work with ST7789.

**Reply by h5n1xp** [reply]
Ok so 40000000 didn't work (just a blank screen), but 39500000 works fine, with only a slight appreciable increase in speed over the previous bus speed (my guess is we are hitting the limit of the psram where the canvas framebuffer  is stored.
